crates/cxx-qt-build/src/lib.rs

```diff
@@ -730,6 +751,35 @@ impl CxxQtBuilder {
 
             let mut qml_metatypes_json = Vec::new();
 
+            // Check that all rust files are within the same directory
+            //
+            // Note we need to do this as moc generates an inputFile which only
+            // includes the file name, qmltyperegistrar then uses this for the
+            // include path (and doesn't consider any prefix).
+            //
+            // This can also be observed when using qt_add_qml_module, if a class
+            // has a QML_ELEMENT the file must be in the same directory as the
+            // CMakeLists and cannot be a relative path to a sub directory.
+            let dirs = qml_module
+                .rust_files
+                .iter()
+                .map(|file| {
+                    file.parent()
+                        .unwrap()
```

Comment:
Prefer actually handling the error by providing a default.

It's theoretically okay if the user inputs e.g. "my_file.rs".
Even though it's probably not a real file, we also shouldn't just panic without explanation.
At least prefer using `expect` with a custom error message.

Reply:
i'll try falling back to an empty string if there is no parent so that in theory `"my_file.rs"` could work